narwhals/_pandas_like/dataframe.py

```diff
@@ -245,10 +248,36 @@ def __getitem__(
     def columns(self) -> list[str]:
         return self._native_frame.columns.tolist()  # type: ignore[no-any-return]
 
+    @overload
+    def rows(
+        self,
+        *,
+        named: Literal[True],
+    ) -> list[dict[str, Any]]: ...
+
+    @overload
+    def rows(
+        self,
+        *,
+        named: Literal[False] = False,
+    ) -> list[tuple[Any, ...]]: ...
+
+    @overload
+    def rows(
+        self,
+        *,
+        named: bool,
+    ) -> list[tuple[Any, ...]] | list[dict[str, Any]]: ...
+
     def rows(
         self, *, named: bool = False
     ) -> list[tuple[Any, ...]] | list[dict[str, Any]]:
         if not named:
+            # cuDF does not support itertuples. But it does support to_dict!
+            if is_cudf_dataframe(self._native_frame):
+                # Extract the row values from the named rows
+                return [tuple(row.values()) for row in self.rows(named=True)]
```

Comment:
You can use `_implementation` attribute instead of directly checking the native frame here. For coverage, let's just tag this branch as no cover:

```suggestion
            if self._implementation is Implementation.CUDF:  # pragma: no cover
                # Extract the row values from the named rows
                return [tuple(row.values()) for row in self.rows(named=True)]
```

Remark that this should affect imports on top of the file